script/tool/lib/src/readme_check_command.dart

```diff
@@ -52,12 +52,20 @@ class ReadmeCheckCommand extends PackageLoopingCommand {
   @override
   bool get hasLongOutput => false;
 
+  @override
+  PackageLoopingType get packageLoopingType =>
+      PackageLoopingType.includeExamples;
+
   @override
   Future<PackageResult> runForPackage(RepositoryPackage package) async {
     final File readme = package.readmeFile;
 
     if (!readme.existsSync()) {
```

Comment:
I've been looking at the way pub.dev handles some of our packages, and I think this test behavior doesn't exactly map to what pub.dev will do. It seems to me that:

* If the package has multiple examples inside `example`, pub will render `example/README.md`, see: [rfw](https://pub.dev/packages/rfw/example)
* If the package has an example directly underneath `example`, they'll render only `example/lib/main.dart` (even if README.md exists), see: [video_player_web](https://pub.dev/packages/video_player_web/example)

In this case we're only checking the actual directories that contain example apps, but we're not looking at the `examples` directory itself in the case we have multiple examples.

I think that in that case, missing the `README.md` should be an error (but that directory is definitely *not* a RepositoryPackage)

Reply:
Oh wow, the precedence of files is quite well described in the [docs](https://dart.dev/tools/pub/package-layout#examples)!

Reply:
I forgot about the `rfw/example/README` case. I can rework this so that we're only looping over top-level packages, and then within them validate example READMEs. I didn't want to do that when it seemed cleaner to change the package looping, since that's the whole point of the package looping command, but as you point out it doesn't quite work this way.

Reply:
I think this is much better than having nothing, and `rfw` is definitely not the most representative of all of the packages we handle. This should totally be merged :P

Reply:
`rfw` is one of the packages we are supposed to handle though, so I fixed it :)